.github/workflows/test-unreleased.yml

```diff
@@ -14,37 +14,68 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        entry:
-          - { opensearch_ref: '1.x' }
-          - { opensearch_ref: '2.x' }
-          - { opensearch_ref: '2.0' }
-          - { opensearch_ref: 'main' }
+        opensearch_ref: ['1.x', '2.x', '2.0', 'main']
+        ruby_version: ['3.1']
```

Comment:
I think you should put back the matrix and add a ruby version field to it because the moment we add another ruby it will explode. Generally that's just unnecessary.